In [1]:
import pandas as pd

Importamos la base de datos de Ogilvy.

In [2]:
df = pd.read_excel('ogilvy.xlsx')  

La desplegamos y mostramos todas las columnas.

In [3]:
pd.options.display.max_columns = None

In [4]:
df.head(5)

Date                                              Title  \
0  2021-02-02 02:26:14.000  @moliverag Recuerdo los formularios de transpa...   
1  2021-02-01 18:29:25.000  Preseleccion olímpica semi fondo y fondo tokio...   
2  2021-02-01 17:19:19.000  RT @PorvenirOficial Empieza el día, la semana ...   
3  2021-02-01 00:05:05.000  Los cambios son p hundirnos + en la pobreza. E...   
4  2021-01-31 16:55:40.000  Que tristeza da saber que la corrupción hace d...   

                                             Snippet  \
0  @moliverag Recuerdo los formularios de transpa...   
1  Preseleccion olímpica semi fondo y fondo tokio...   
2  RT @PorvenirOficial Empieza el día, la semana ...   
3  Los cambios son p hundirnos + en la pobreza. E...   
4  Que tristeza da saber que la corrupción hace d...   

                                                 Url       Domain Sentiment  \
0  http://twitter.com/delioquintero1/statuses/135...  twitter.com  negative   
1  http://twitter.com/gerardgiraldo/statuses/1356...  twitter.com   neutral   
2  http://twitter.com/jjcastro_v/statuses/1356291...  twitter.com   neutral   
3  http://twitter.com/delioquintero1/statuses/135...  twitter.com  negative   
4  http://twitter.com/delioquintero1/statuses/135...  twitter.com  negative   

   Emotion Page Type Language Country Code Continent Code      Continent  \
0      NaN   twitter       es          COL  SOUTH AMERICA  South America   
1      NaN   twitter       es          COL  SOUTH AMERICA  South America   
2      NaN   twitter       es          COL  SOUTH AMERICA  South America   
3      NaN   twitter       es          COL  SOUTH AMERICA  South America   
4      NaN   twitter       es          COL  SOUTH AMERICA  South America   

    Country Region Code  Region          City Code Account Type  Assignment  \
0  Colombia         NaN     NaN                NaN   individual         NaN   
1  Colombia  COL.Bogota  Bogota  COL.Bogota.Bogota   individual         NaN   
2  Colombia  COL.Bogota  Bogota  COL.Bogota.Bogota   individual         NaN   
3  Colombia         NaN     NaN                NaN   individual         NaN   
4  Colombia         NaN     NaN                NaN   individual         NaN   

           Author                                             Avatar  \
0  delioquintero1  https://audiences.brandwatch.com/api/audiences...   
1   gerardgiraldo  https://audiences.brandwatch.com/api/audiences...   
2      jjcastro_v  https://audiences.brandwatch.com/api/audiences...   
3  delioquintero1  https://audiences.brandwatch.com/api/audiences...   
4  delioquintero1  https://audiences.brandwatch.com/api/audiences...   

   Category Details  Checked    City  Display URLs  \
0               NaN    False     NaN           NaN   
1               NaN    False  Bogota           NaN   
2               NaN    False  Bogota           NaN   
3               NaN    False     NaN           NaN   
4               NaN    False     NaN           NaN   

                                       Expanded URLs  Facebook Author ID  \
0                                                NaN                 NaN   
1  https://www.instagram.com/p/CKwnpKJD5x3/?igshi...                 NaN   
2        https://www.youtube.com/watch?v=r5atUZbDgMo                 NaN   
3                                                NaN                 NaN   
4                                                NaN                 NaN   

   Facebook Comments  Facebook Likes  Facebook Role  Facebook Shares  \
0                  0               0            NaN                0   
1                  0               0            NaN                0   
2                  0               0            NaN                0   
3                  0               0            NaN                0   
4                  0               0            NaN                0   

   Facebook Subtype                         Full Name Gender  \
0               NaN   delioquintero1 (delio quintero)   male   
1              

Confirmamos que solo hay datos de Twitter en esta base de datos.

In [7]:
df["Page Type"].unique()

array(['twitter'], dtype=object)

Creamos una copia para no alterar la base de datos original.

In [41]:
dfcopy = df.copy()

In [42]:
dfcopy.head()

Date                                              Title  \
0  2021-02-02 02:26:14.000  @moliverag Recuerdo los formularios de transpa...   
1  2021-02-01 18:29:25.000  Preseleccion olímpica semi fondo y fondo tokio...   
2  2021-02-01 17:19:19.000  RT @PorvenirOficial Empieza el día, la semana ...   
3  2021-02-01 00:05:05.000  Los cambios son p hundirnos + en la pobreza. E...   
4  2021-01-31 16:55:40.000  Que tristeza da saber que la corrupción hace d...   

                                             Snippet  \
0  @moliverag Recuerdo los formularios de transpa...   
1  Preseleccion olímpica semi fondo y fondo tokio...   
2  RT @PorvenirOficial Empieza el día, la semana ...   
3  Los cambios son p hundirnos + en la pobreza. E...   
4  Que tristeza da saber que la corrupción hace d...   

                                                 Url       Domain Sentiment  \
0  http://twitter.com/delioquintero1/statuses/135...  twitter.com  negative   
1  http://twitter.com/gerardgiraldo/statuses/1356...  twitter.com   neutral   
2  http://twitter.com/jjcastro_v/statuses/1356291...  twitter.com   neutral   
3  http://twitter.com/delioquintero1/statuses/135...  twitter.com  negative   
4  http://twitter.com/delioquintero1/statuses/135...  twitter.com  negative   

   Emotion Page Type Language Country Code Continent Code      Continent  \
0      NaN   twitter       es          COL  SOUTH AMERICA  South America   
1      NaN   twitter       es          COL  SOUTH AMERICA  South America   
2      NaN   twitter       es          COL  SOUTH AMERICA  South America   
3      NaN   twitter       es          COL  SOUTH AMERICA  South America   
4      NaN   twitter       es          COL  SOUTH AMERICA  South America   

    Country Region Code  Region          City Code Account Type  Assignment  \
0  Colombia         NaN     NaN                NaN   individual         NaN   
1  Colombia  COL.Bogota  Bogota  COL.Bogota.Bogota   individual         NaN   
2  Colombia  COL.Bogota  Bogota  COL.Bogota.Bogota   individual         NaN   
3  Colombia         NaN     NaN                NaN   individual         NaN   
4  Colombia         NaN     NaN                NaN   individual         NaN   

           Author                                             Avatar  \
0  delioquintero1  https://audiences.brandwatch.com/api/audiences...   
1   gerardgiraldo  https://audiences.brandwatch.com/api/audiences...   
2      jjcastro_v  https://audiences.brandwatch.com/api/audiences...   
3  delioquintero1  https://audiences.brandwatch.com/api/audiences...   
4  delioquintero1  https://audiences.brandwatch.com/api/audiences...   

   Category Details  Checked    City  Display URLs  \
0               NaN    False     NaN           NaN   
1               NaN    False  Bogota           NaN   
2               NaN    False  Bogota           NaN   
3               NaN    False     NaN           NaN   
4               NaN    False     NaN           NaN   

                                       Expanded URLs  Facebook Author ID  \
0                                                NaN                 NaN   
1  https://www.instagram.com/p/CKwnpKJD5x3/?igshi...                 NaN   
2        https://www.youtube.com/watch?v=r5atUZbDgMo                 NaN   
3                                                NaN                 NaN   
4                                                NaN                 NaN   

   Facebook Comments  Facebook Likes  Facebook Role  Facebook Shares  \
0                  0               0            NaN                0   
1                  0               0            NaN                0   
2                  0               0            NaN                0   
3                  0               0            NaN                0   
4                  0               0            NaN                0   

   Facebook Subtype                         Full Name Gender  \
0               NaN   delioquintero1 (delio quintero)   male   
1              

Se puede eliminar las menciones repetidas desde 2 puntos de vista:
En el primero eliminamos todos los RTs y tuits repetidos de nuestra base de datos; esto para que no se tengan tuits repetidos donde los usuarios tengan las mismas menciones. Filtramos por Title del tuit, donde esta el contenido que se repite.

In [13]:
df2 = dfcopy.drop_duplicates(subset=["Title"], keep=False)


In [15]:
pd.options.display.max_rows = None

In [16]:
df2.head(200)

Date  \
0    2021-02-02 02:26:14.000   
1    2021-02-01 18:29:25.000   
2    2021-02-01 17:19:19.000   
3    2021-02-01 00:05:05.000   
4    2021-01-31 16:55:40.000   
5    2021-01-31 16:36:52.000   
6    2021-01-30 21:07:10.000   
7    2021-01-29 19:32:55.000   
8    2021-01-29 19:22:34.000   
27   2021-01-28 17:25:48.000   
29   2021-01-28 17:25:40.000   
30   2021-01-28 17:00:26.000   
31   2021-01-28 16:48:31.000   
32   2021-01-28 16:44:48.000   
33   2021-01-28 15:56:20.000   
34   2021-01-28 15:21:20.000   
35   2021-01-28 15:02:21.000   
36   2021-01-28 15:01:44.000   
37   2021-01-28 15:00:41.000   
38   2021-01-28 15:00:02.000   
39   2021-01-27 23:04:48.000   
40   2021-01-27 21:20:41.000   
41   2021-01-27 19:20:58.000   
42   2021-01-27 19:18:16.000   
43   2021-01-27 18:59:31.000   
44   2021-01-27 18:10:01.000   
45   2021-01-27 18:06:43.000   
46   2021-01-27 17:37:43.000   
47   2021-01-27 17:29:32.000   
48   2021-01-27 17:18:40.000   
49   2021-01-27 17:15:02.000   
50   2021-01-27 17:14:13.000   
51   2021-01-27 17:12:39.000   
52   2021-01-27 17:04:31.000   
53   2021-01-27 16:10:08.000   
54   2021-01-27 15:48:49.000   
55   2021-01-27 15:45:45.000   
59   2021-01-27 15:24:26.000   
60   2021-01-27 15:19:44.000   
61   2021-01-27 14:46:17.000   
62   2021-01-27 14:46:09.000   
63   2021-01-27 14:46:08.000   
64   2021-01-26 22:50:22.000   
65   2021-01-26 21:45:53.000   
66   2021-01-26 21:35:30.000   
67   2021-01-26 21:34:45.000   
68   2021-01-26 19:45:24.000   
70   2021-01-26 18:43:52.000   
71   2021-01-26 18:43:51.000   
72   2021-01-26 18:20:23.000   
73   2021-01-26 18:13:09.000   
74   2021-01-26 16:55:36.000   
75   2021-01-26 16:55:12.000   
76   2021-01-26 16:55:09.000   
77   2021-01-26 16:55:03.000   
78   2021-01-26 01:27:53.000   
80   2021-01-26 00:40:27.000   
81   2021-01-26 00:40:00.000   
82   2021-01-25 23:29:21.000   
83   2021-01-25 22:16:53.000   
85   2021-01-25 21:40:44.000   
86   2021-01-25 20:08:38.000   
87   2021-01-25 17:50:20.000   
88   2021-01-25 17:50:14.000   
89   2021-01-25 17:50:13.000   
90   2021-01-25 15:54:31.000   
91   2021-01-25 15:28:21.000   
92   2021-01-25 15:20:23.000   
93   2021-01-24 21:45:14.000   
94   2021-01-24 17:45:19.000   
95   2021-01-24 14:51:41.000   
96   2021-01-24 04:01:19.000   
99   2021-01-22 23:50:09.000   
100  2021-01-22 22:04:08.000   
102  2021-01-22 20:47:09.000   
103  2021-01-22 19:09:19.000   
105  2021-01-22 17:22:39.000   
108  2021-01-22 17:13:07.000   
110  2021-01-22 16:56:40.000   
111  2021-01-22 16:51:01.000   
112  2021-01-22 15:56:10.000   
113  2021-01-22 15:18:02.000   
114  2021-01-22 01:48:01.000   
115  2021-01-22 01:23:15.000   
116  2021-01-21 20:09:46.000   
117  2021-01-21 17:06:22.000   
118  2021-01-21 17:06:20.000   
119  2021-01-21 16:38:25.000   
120  2021-01-21 16:31:29.000   
121  2021-01-21 16:26:59.000   
122  2021-01-21 16:26:57.000   
123  2021-01-21 15:55:44.000   
125  2021-01-20 23:41:58.000   
126  2021-01-20 23:14:22.000   
128  2021-01-20 22:24:52.000   
129  2021-01-20 22:24:28.000   
131  2021-01-20 22:17:42.000   
133  2021-01-20 22:04:45.000   
136  2021-01-20 19:46:20.000   
138  2021-01-20 19:41:22.000   
139  2021-01-20 16:14:08.000   
140  2021-01-20 15:58:57.000   
141  2021-01-20 15:58:49.000   
142  2021-01-20 15:57:05.000   
143  2021-01-20 15:55:22.000   
144  2021-01-20 15:55:21.000   
145  2021-01-20 12:29:04.000   
147  2021-01-19 23:24:23.000   
148  2021-01-19 22:17:50.000   
149  2021-01-19 18:02:46.000   
150  2021-01-19 17:49:14.000   
151  2021-01-19 17:41:32.000   
152  2021-01-19 17:32:36.000   
153  2021-01-19 17:32:35.000   
154  2021-01-19 17:32:33.000   
156  2021-01-19 17:20:33.000   
157  2021-01-19 17:03:20.000   
158  2021-01-19 16:54:59.000   
159  2021-01-19 12:27:14.000   
160  2021-01-19 12:26:23.000   
161  2021-01-18 21:51:30.000   
162  2021-01-18 21:41:20.000   
163  2021-01-18 21:41:19.000   
164  2021-01-18 19:46:11.000   
165  2021-01-18 17:13:43

Descartando retuits y tuits repetidos obtenemos que se tienen 2635 menciones. Cabe resaltar que se realizó una verificación para validar que no esta el tuit original en toda la base de datos del retuit que queda, por lo tanto el contenido del RT no se repite.

In [21]:
df2.shape[0]

2635

In [23]:
dfcopy[dfcopy["Title"].str.contains("@gurozu")]

Date  \
10    2021-01-29 17:16:32.000   
11    2021-01-29 17:15:42.000   
69    2021-01-26 19:06:33.000   
73    2021-01-26 18:13:09.000   
78    2021-01-26 01:27:53.000   
79    2021-01-26 01:27:19.000   
99    2021-01-22 23:50:09.000   
108   2021-01-22 17:13:07.000   
526   2020-12-12 17:48:26.000   
527   2020-12-12 17:13:50.000   
3712  2020-10-18 01:47:26.000   
3714  2020-10-17 19:50:25.000   
5001  2021-01-29 17:16:32.000   
5002  2021-01-29 17:15:42.000   

                                                  Title  \
10    RT @gurozu Ojo con fondos privados de pensione...   
11    RT @gurozu Ojo con fondos privados de pensione...   
69    RT @gurozu Ojo con fondos privados de pensione...   
73    @CorinneLover @gurozu @proteccion @PorvenirOfi...   
78    @gurozu @proteccion @PorvenirOficial Para que ...   
79    RT @gurozu Ojo con fondos privados de pensione...   
99    @gurozu @proteccion @PorvenirOficial Y además ...   
108   RT @gurozu Ojo con fondos privados de pensione...   
526   RT @delioquintero1 @mancchozongo1 @merv59 @mac...   
527   @mancchozongo1 @merv59 @macdonal5 @PGN_COL @fc...   
3712  RT @delioquintero1 @macdonal5 @CorteSupremaJ @...   
3714  @macdonal5 @CorteSupremaJ @CConstitucional @co...   
5001  RT @gurozu Ojo con fondos privados de pensione...   
5002  RT @gurozu Ojo con fondos privados de pensione...   

                                                Snippet  \
10    RT @gurozu Ojo con fondos privados de pensione...   
11    RT @gurozu Ojo con fondos privados de pensione...   
69    RT @gurozu Ojo con fondos privados de pensione...   
73    @CorinneLover @gurozu @proteccion @PorvenirOfi...   
78    @gurozu @proteccion @PorvenirOficial Para que ...   
79    RT @gurozu Ojo con fondos privados de pensione...   
99    @gurozu @proteccion @PorvenirOficial Y además ...   
108   RT @gurozu Ojo con fondos privados de pensione...   
526   RT @delioquintero1 @mancchozongo1 @merv59 @mac...   
527   @mancchozongo1 @merv59 @macdonal5 @PGN_COL @fc...   
3712  ...denegar justicia en fraude procesal de @Por...   
3714  ...fraude procesal de @PorvenirOficial-TRANSUR...   
5001  RT @gurozu Ojo con fondos privados de pensione...   
5002  RT @gurozu Ojo con fondos privados de pensione...   

                                                    Url       Domain  \
10    http://twitter.com/psandresg/statuses/13552031...  twitter.com   
11    http://twitter.com/FGVz_73/statuses/1355202976...  twitter.com   
69    http://twitter.com/MOTORESYLEYES/statuses/1354...  twitter.com   
73    http://twitter.com/alejomoreno1972/statuses/13...  twitter.com   
78    http://twitter.com/alejomoreno1972/statuses/13...  twitter.com   
79    http://twitter.com/alejomoreno1972/statuses/13...  twitter.com   
99    http://twitter.com/JavierAdaime/statuses/13527...  twitter.com   
108   http://twitter.com/RubenSelvaYMar/statuses/135...  twitter.com   
526   http://twitter.com/macdonal5/statuses/13378165...  twitter.com   
527   http://twitter.com/delioquintero1/statuses/133...  twitter.com   
3712  http://twitter.com/macdonal5/statuses/13176434...  twitter.com   
3714  http://twitter.com/delioquintero1/statuses/131...  twitter.com   
5001  http://twitter.com/psandresg/statuses/13552031...  twitter.com   
5002  http://twitter.com/FGVz_73/statuses/1355202976...  twitter.com   

     Sentiment  Emotion Page Type Language Country Code Continent Code  \
10    negative      NaN   twitter       es          COL  SOUTH AMERICA   
11    negative      NaN   twitter       es          COL  SOUTH AMERICA   
69    negative      NaN   twitter       es          COL  SOUTH AMERICA   
73     neutral      NaN   twitter       es          COL  SOUTH AMERICA   
78     neutral      NaN   twitter       es          COL  SOUTH AMERICA   
79    negative      NaN   twitter       es          COL  SOUTH AMERICA   
99    negative      NaN   twitter       es          COL  SOUTH AMERICA   
108   negative      NaN   twitter       es          COL  SOUTH AMERICA   
526   nega

In [24]:
df2[df2["Title"].str.contains("@gurozu")]

Date  \
73    2021-01-26 18:13:09.000   
78    2021-01-26 01:27:53.000   
99    2021-01-22 23:50:09.000   
108   2021-01-22 17:13:07.000   
526   2020-12-12 17:48:26.000   
527   2020-12-12 17:13:50.000   
3712  2020-10-18 01:47:26.000   
3714  2020-10-17 19:50:25.000   

                                                  Title  \
73    @CorinneLover @gurozu @proteccion @PorvenirOfi...   
78    @gurozu @proteccion @PorvenirOficial Para que ...   
99    @gurozu @proteccion @PorvenirOficial Y además ...   
108   RT @gurozu Ojo con fondos privados de pensione...   
526   RT @delioquintero1 @mancchozongo1 @merv59 @mac...   
527   @mancchozongo1 @merv59 @macdonal5 @PGN_COL @fc...   
3712  RT @delioquintero1 @macdonal5 @CorteSupremaJ @...   
3714  @macdonal5 @CorteSupremaJ @CConstitucional @co...   

                                                Snippet  \
73    @CorinneLover @gurozu @proteccion @PorvenirOfi...   
78    @gurozu @proteccion @PorvenirOficial Para que ...   
99    @gurozu @proteccion @PorvenirOficial Y además ...   
108   RT @gurozu Ojo con fondos privados de pensione...   
526   RT @delioquintero1 @mancchozongo1 @merv59 @mac...   
527   @mancchozongo1 @merv59 @macdonal5 @PGN_COL @fc...   
3712  ...denegar justicia en fraude procesal de @Por...   
3714  ...fraude procesal de @PorvenirOficial-TRANSUR...   

                                                    Url       Domain  \
73    http://twitter.com/alejomoreno1972/statuses/13...  twitter.com   
78    http://twitter.com/alejomoreno1972/statuses/13...  twitter.com   
99    http://twitter.com/JavierAdaime/statuses/13527...  twitter.com   
108   http://twitter.com/RubenSelvaYMar/statuses/135...  twitter.com   
526   http://twitter.com/macdonal5/statuses/13378165...  twitter.com   
527   http://twitter.com/delioquintero1/statuses/133...  twitter.com   
3712  http://twitter.com/macdonal5/statuses/13176434...  twitter.com   
3714  http://twitter.com/delioquintero1/statuses/131...  twitter.com   

     Sentiment  Emotion Page Type Language Country Code Continent Code  \
73     neutral      NaN   twitter       es          COL  SOUTH AMERICA   
78     neutral      NaN   twitter       es          COL  SOUTH AMERICA   
99    negative      NaN   twitter       es          COL  SOUTH AMERICA   
108   negative      NaN   twitter       es          COL  SOUTH AMERICA   
526   negative      NaN   twitter       es          COL  SOUTH AMERICA   
527   negative      NaN   twitter       es          COL  SOUTH AMERICA   
3712  negative      NaN   twitter       es          COL  SOUTH AMERICA   
3714  negative      NaN   twitter       es          COL  SOUTH AMERICA   

          Continent   Country Region Code  Region          City Code  \
73    South America  Colombia  COL.Bogota  Bogota  COL.Bogota.Bogota   
78    South America  Colombia  COL.Bogota  Bogota  COL.Bogota.Bogota   
99    South America  Colombia  COL.Bogota  Bogota  COL.Bogota.Bogota   
108   South America  Colombia  COL.Bogota  Bogota  COL.Bogota.Bogota   
526   South America  Colombia  COL.Bogota  Bogota  COL.Bogota.Bogota   
527   South America  Colombia         NaN     NaN                NaN   
3712  South America  Colombia  COL.Bogota  Bogota  COL.Bogota.Bogota   
3714  South America  Colombia         NaN     NaN                NaN   

     Account Type  Assignment           Author  \
73     individual         NaN  alejomoreno1972   
78     individual         NaN  alejomoreno1972   
99     individual         NaN     JavierAdaime   
108    individual         NaN   RubenSelvaYMar   
526    individual         NaN        macdonal5   
527    individual         NaN   delioquintero1   
3712   individual         NaN        macdonal5   
3714   individual         NaN   delioquintero1   

                                                 Avatar  Category Details  \
73    https://audiences.brandwatch.com/api/audiences...               NaN   
78    https://audiences.brandwatch.com/api/audiences...               NaN   
99    https://au

También podemos realizar esta limpieza desde otra perspectiva. Donde filtramos por autor del tuit o retuit y solo tenemos en cuenta una vez a dicho autor en toda la base de datos. Esto desde una perspectiva de negocio es mas viable debido a que con el numero total de personas que estan mencionando nuestra marca y la cantidad de seguidores que poseen, ya sea con #, @ o simplemente hablando sobre ella, podemos calcular el alcance potencial.

In [43]:
df3 = dfcopy.drop_duplicates(subset=["Author"], keep=False)

In [44]:
df3.head(100)

Date  \
8    2021-01-29 19:22:34.000   
45   2021-01-27 18:06:43.000   
49   2021-01-27 17:15:02.000   
56   2021-01-27 15:30:57.000   
57   2021-01-27 15:29:56.000   
69   2021-01-26 19:06:33.000   
82   2021-01-25 23:29:21.000   
85   2021-01-25 21:40:44.000   
86   2021-01-25 20:08:38.000   
92   2021-01-25 15:20:23.000   
95   2021-01-24 14:51:41.000   
96   2021-01-24 04:01:19.000   
99   2021-01-22 23:50:09.000   
126  2021-01-20 23:14:22.000   
127  2021-01-20 23:01:16.000   
134  2021-01-20 20:55:41.000   
137  2021-01-20 19:43:26.000   
145  2021-01-20 12:29:04.000   
148  2021-01-19 22:17:50.000   
159  2021-01-19 12:27:14.000   
160  2021-01-19 12:26:23.000   
167  2021-01-18 16:19:45.000   
168  2021-01-18 15:20:52.000   
182  2021-01-15 19:22:00.000   
191  2021-01-14 17:28:42.000   
195  2021-01-14 12:45:20.000   
213  2021-01-13 14:54:48.000   
214  2021-01-13 04:51:57.000   
218  2021-01-12 23:13:23.000   
224  2021-01-12 16:47:58.000   
228  2021-01-12 16:18:38.000   
233  2021-01-11 21:21:35.000   
235  2021-01-11 03:49:43.000   
237  2021-01-11 01:52:41.000   
239  2021-01-10 13:33:41.000   
240  2021-01-09 23:00:29.000   
241  2021-01-09 21:28:43.000   
242  2021-01-09 21:22:26.000   
246  2021-01-08 21:34:13.000   
250  2021-01-08 20:24:24.000   
267  2021-01-07 16:06:21.000   
280  2021-01-06 16:10:42.000   
285  2021-01-05 21:21:50.000   
291  2021-01-05 16:48:57.000   
333  2020-12-30 01:38:49.000   
343  2020-12-29 04:10:52.000   
345  2020-12-28 20:30:15.000   
346  2020-12-28 20:28:36.000   
347  2020-12-28 20:17:29.000   
351  2020-12-28 16:03:57.000   
353  2020-12-25 23:43:58.000   
359  2020-12-23 19:39:06.000   
360  2020-12-23 19:21:31.000   
364  2020-12-23 16:22:18.000   
365  2020-12-23 16:20:30.000   
369  2020-12-23 00:13:01.000   
374  2020-12-22 18:15:19.000   
375  2020-12-22 18:10:05.000   
393  2020-12-20 21:25:58.000   
394  2020-12-20 20:04:07.000   
395  2020-12-20 17:32:54.000   
398  2020-12-19 17:52:45.000   
400  2020-12-19 02:05:38.000   
409  2020-12-18 20:57:50.000   
419  2020-12-17 23:05:00.000   
422  2020-12-17 21:12:10.000   
436  2020-12-17 03:26:21.000   
438  2020-12-16 21:53:01.000   
443  2020-12-16 16:40:34.000   
454  2020-12-15 17:29:02.000   
462  2020-12-15 10:59:22.000   
467  2020-12-14 22:40:15.000   
503  2020-12-13 18:33:15.000   
504  2020-12-13 18:31:19.000   
505  2020-12-13 16:02:16.000   
509  2020-12-13 06:43:24.000   
515  2020-12-13 04:19:14.000   
524  2020-12-13 01:48:48.000   
525  2020-12-12 22:07:31.000   
528  2020-12-12 16:11:26.000   
530  2020-12-12 15:05:13.000   
531  2020-12-12 12:37:41.000   
542  2020-12-11 12:53:25.000   
558  2020-12-10 00:07:05.000   
559  2020-12-09 22:37:56.000   
568  2020-12-09 10:56:50.000   
570  2020-12-08 19:19:03.000   
633  2020-12-06 16:55:54.000   
634  2020-12-06 16:19:18.000   
635  2020-12-06 15:33:53.000   
639  2020-12-06 15:20:29.000   
661  2020-12-05 20:06:51.000   
665  2020-12-05 13:59:57.000   
684  2020-12-03 12:44:54.000   
685  2020-12-03 10:58:10.000   
697  2020-12-02 16:52:43.000   
698  2020-12-02 16:52:35.000   
700  2020-12-02 14:40:53.000   
713  2020-12-01 15:35:45.000   
723  2020-11-30 18:35:18.000   

                                                 Title  \
8    Me llegan unos extractos de  @PorvenirOficial ...   
45   RT @smarco943 Un fondo de pension me puede tra...   
49   Que buen noticia para Colombia! Y qué dice al ...   
56   RT @SaraAcostaM Me enteré de que también estoy...   
57   RT @SaraAcostaM Me enteré de que también estoy...   
69   RT @gurozu Ojo con fondos privados de pensione...   
82   RT @DeBalbinV En 30 años @proteccion @SkandiaC...   
85   Estamos en medio de una emergencia sanitaria y...   
86   En 30 años @proteccion @SkandiaCol @ColfondosO...   
92   Que debo hacer para ser un corredor externo de...   
95   Señores @PorvenirOficial ¿Por donde se radica ...   
96   @meriemorenita @CurrumeoMD Yo si estoy en @Por...   
99   @gurozu @proteccion @PorvenirOfi

In [49]:
usuarios_mencionando = df3.shape[0]
print(usuarios_mencionando)

1544


Por lo tanto tenemos que hay 1544 usuarios mencionando o retuiteando a nuestra marca en sus tuits. 

Para calcular el alcance potencial, tenemos en cuenta que se suma el número de usuarios que están mencionando tu marca o que retuitean una publicación de tu perfil + el número de seguidores que tienen estos usuarios.

In [51]:
seguidores_totales = df3["Twitter Followers"].sum()
print(seguidores_totales)

11538568


In [53]:
Alcance_pot = usuarios_mencionando + seguidores_totales
print(Alcance_pot)

11540112


Nuestro alcance potencial son 11.540.112 usuarios.

Ahora tenemos en cuenta, que para calcular la cantidad total de interacciones que hay en nuestra base de datos, hay que sumar los clics en cualquier lugar del tuit, ya sea retuits, respuestas, seguimientos, Me gusta, enlaces, cards, hashtags, contenido multimedia insertado, nombre de usuario, foto de perfil o ampliación del tuit. 
Pero en esta base de datos en particular vamos a tener en cuenta solo las columnas: 
- "Twitter Reply Count"(cantidad de veces que respondieron un tweet o retuit)	
- "Twitter Reply to" (Indica que el tuit o retuit en cuestion fue una respuesta a otro tuit)
- "Twitter Retweet of" (Indica que el tuit en cuestión es un retuit de otro)
- "Twitter Retweets" (Indica el numero de retuits que se le hicieron al tuit en cuestión)
En este punto tenemos en cuenta la base de datos total, ya que las interacciones totales es la suma de las acciones anteriormente realizadas en cada uno de los tuits.

In [54]:
dfcopy["Twitter Reply to"].count()

1427

In [119]:
TwReplyCount = dfcopy["Twitter Reply Count"].sum()
TwReplyto = dfcopy["Twitter Reply to"].count()
TwRtof = dfcopy["Twitter Retweet of"].count()
TwRt = dfcopy["Twitter Retweets"].sum()
Interacciones = TwReplyCount + TwReplyto + TwRtof + TwRt
print(Interacciones)
print(TwReplyCount, TwReplyto, TwRtof, TwRt)

10756
1849 1427 3076 4404


Se tiene un total de 10756 interacciones en la base de datos teniendo en cuenta tuits y retuits.

Ahora para calcular el día donde se recibieron mas menciones debemos separar la columna date en un formato de fecha día y otro donde este la hora.

In [61]:
dfcopy["Dates"] = pd.to_datetime(dfcopy["Date"]).dt.date
dfcopy["Time"] = pd.to_datetime(dfcopy["Date"]).dt.time

In [62]:
dfcopy.head()

Date                                              Title  \
0  2021-02-02 02:26:14.000  @moliverag Recuerdo los formularios de transpa...   
1  2021-02-01 18:29:25.000  Preseleccion olímpica semi fondo y fondo tokio...   
2  2021-02-01 17:19:19.000  RT @PorvenirOficial Empieza el día, la semana ...   
3  2021-02-01 00:05:05.000  Los cambios son p hundirnos + en la pobreza. E...   
4  2021-01-31 16:55:40.000  Que tristeza da saber que la corrupción hace d...   

                                             Snippet  \
0  @moliverag Recuerdo los formularios de transpa...   
1  Preseleccion olímpica semi fondo y fondo tokio...   
2  RT @PorvenirOficial Empieza el día, la semana ...   
3  Los cambios son p hundirnos + en la pobreza. E...   
4  Que tristeza da saber que la corrupción hace d...   

                                                 Url       Domain Sentiment  \
0  http://twitter.com/delioquintero1/statuses/135...  twitter.com  negative   
1  http://twitter.com/gerardgiraldo/statuses/1356...  twitter.com   neutral   
2  http://twitter.com/jjcastro_v/statuses/1356291...  twitter.com   neutral   
3  http://twitter.com/delioquintero1/statuses/135...  twitter.com  negative   
4  http://twitter.com/delioquintero1/statuses/135...  twitter.com  negative   

   Emotion Page Type Language Country Code Continent Code      Continent  \
0      NaN   twitter       es          COL  SOUTH AMERICA  South America   
1      NaN   twitter       es          COL  SOUTH AMERICA  South America   
2      NaN   twitter       es          COL  SOUTH AMERICA  South America   
3      NaN   twitter       es          COL  SOUTH AMERICA  South America   
4      NaN   twitter       es          COL  SOUTH AMERICA  South America   

    Country Region Code  Region          City Code Account Type  Assignment  \
0  Colombia         NaN     NaN                NaN   individual         NaN   
1  Colombia  COL.Bogota  Bogota  COL.Bogota.Bogota   individual         NaN   
2  Colombia  COL.Bogota  Bogota  COL.Bogota.Bogota   individual         NaN   
3  Colombia         NaN     NaN                NaN   individual         NaN   
4  Colombia         NaN     NaN                NaN   individual         NaN   

           Author                                             Avatar  \
0  delioquintero1  https://audiences.brandwatch.com/api/audiences...   
1   gerardgiraldo  https://audiences.brandwatch.com/api/audiences...   
2      jjcastro_v  https://audiences.brandwatch.com/api/audiences...   
3  delioquintero1  https://audiences.brandwatch.com/api/audiences...   
4  delioquintero1  https://audiences.brandwatch.com/api/audiences...   

   Category Details  Checked    City  Display URLs  \
0               NaN    False     NaN           NaN   
1               NaN    False  Bogota           NaN   
2               NaN    False  Bogota           NaN   
3               NaN    False     NaN           NaN   
4               NaN    False     NaN           NaN   

                                       Expanded URLs  Facebook Author ID  \
0                                                NaN                 NaN   
1  https://www.instagram.com/p/CKwnpKJD5x3/?igshi...                 NaN   
2        https://www.youtube.com/watch?v=r5atUZbDgMo                 NaN   
3                                                NaN                 NaN   
4                                                NaN                 NaN   

   Facebook Comments  Facebook Likes  Facebook Role  Facebook Shares  \
0                  0               0            NaN                0   
1                  0               0            NaN                0   
2                  0               0            NaN                0   
3                  0               0            NaN                0   
4                  0               0            NaN                0   

   Facebook Subtype                         Full Name Gender  \
0               NaN   delioquintero1 (delio quintero)   male   
1              

In [151]:
dfcopy.groupby('Dates')['Dates'].count().sort_values(ascending=False)

Dates
2020-10-30    502
2020-11-13    499
2020-11-12    475
2020-10-31    421
2020-10-29    347
2020-08-13    240
2020-12-06     75
2020-11-15     71
2020-11-14     65
2020-12-07     62
2020-11-04     55
2020-08-14     55
2020-09-08     49
2020-08-11     46
2020-10-26     45
2020-10-15     44
2021-01-04     42
2020-12-14     37
2020-08-31     35
2020-11-09     33
2020-11-17     32
2020-10-09     32
2020-09-21     31
2020-10-08     31
2020-08-18     30
2020-08-26     29
2021-01-28     28
2020-10-19     27
2020-09-28     27
2020-10-28     26
2021-01-27     25
2020-09-22     25
2020-10-20     24
2020-09-29     24
2020-11-06     24
2020-10-07     23
2020-12-13     23
2020-08-28     23
2020-10-13     23
2020-11-26     23
2020-11-05     23
2020-10-14     23
2020-09-23     23
2020-11-11     23
2020-10-02     22
2020-11-25     22
2020-09-24     22
2021-01-13     22
2020-10-22     22
2021-01-29     22
2020-11-19     22
2020-10-16     21
2021-01-20     21
2020-09-16     21
2020-11-16     20
2020

Se evidencia que el día donde hubo mas actividad fue el 30 de octubre de 2020 con 502 menciones. Evidenciando alta actividad a finales del mes de octubre y principios de noviembre. Al parecer para estas fechas Porvenir auspició una media maratón de la cual se habló en redes sociales, sin olvidar el descontento general de las personas por el servicio prestado por la entidad.

In [108]:
print(dfcopy["Dates"].value_counts().head())


2020-10-30    502
2020-11-13    499
2020-11-12    475
2020-10-31    421
2020-10-29    347
Name: Dates, dtype: int64


gráfica Menciones recibidas vs sentimiento

Siendo el alcance potencial la cantidad de usuarios que siguen al autor de la mención, tenemos que:

In [107]:
Max= dfcopy["Twitter Followers"].max()
print("El número de seguidores es: ", Max)
print("Nombre de usuario: ", dfcopy["Author"].where(dfcopy["Twitter Followers"] == Max).dropna().values[0])
print("Nombre completo: ", dfcopy["Full Name"].where(dfcopy["Twitter Followers"] == Max).dropna().values[0])
print("Fecha: ", dfcopy["Dates"].where(dfcopy["Twitter Followers"] == Max).dropna().values[0])

El número de seguidores es:  4563978
Nombre de usuario:  RevistaSemana
Nombre completo:  RevistaSemana (Revista Semana)
Fecha:  2020-10-22


La mención con mas alcance potencial es esa donde el usuario que la escribió tiene mas seguidores. Por lo tanto tenemos que @RevistaSemana con 4.563.978 seguidores hizo la mención con mayor alcance potencial.
La hizo el 22 de octubre de 2020.

In [115]:
dfcopy[dfcopy["Author"].str.contains("RevistaSemana")]

Date  \
3625  2020-10-22 21:11:47.000   

                                                  Title  \
3625  RT @ForosSemana Miguel Largacha, presidente de...   

                                                Snippet  \
3625  RT @ForosSemana Miguel Largacha, presidente de...   

                                                    Url       Domain  \
3625  http://twitter.com/RevistaSemana/statuses/1319...  twitter.com   

     Sentiment  Emotion Page Type Language Country Code Continent Code  \
3625   neutral      NaN   twitter       es          COL  SOUTH AMERICA   

          Continent   Country Region Code  Region          City Code  \
3625  South America  Colombia  COL.Bogota  Bogota  COL.Bogota.Bogota   

     Account Type  Assignment         Author  \
3625   individual         NaN  RevistaSemana   

                                                 Avatar  Category Details  \
3625  https://audiences.brandwatch.com/api/audiences...               NaN   

      Checked    City  Display URLs  \
3625    False  Bogota           NaN   

                                          Expanded URLs  Facebook Author ID  \
3625  https://www.pscp.tv/w/cmCZKzMzMjQxNzd8MVprS3pt...                 NaN   

      Facebook Comments  Facebook Likes  Facebook Role  Facebook Shares  \
3625                  0               0            NaN                0   

      Facebook Subtype                       Full Name   Gender  \
3625               NaN  RevistaSemana (Revista Semana)  unknown   

                 Hashtags  Impact  Impressions  Instagram Comments  \
3625  #finanzasparalavida    83.7      4563978                   0   

      Instagram Followers  Instagram Following  Instagram Likes  \
3625                    0                    0                0   

      Instagram Posts Interest  Last Assignment Date  Latitude  \
3625                0      NaN                   NaN         0   

            Location Name  Longitude  Media Filter Media URLs  \
3625  Bogota, Bogota, COL          0           NaN        NaN   

                                      Mentioned Authors  Priority Professions  \
3625  @forossemana, @porveniroficial, @revistadinero...       NaN         NaN   

                           Resource Id               Short URLs  Starred  \
3625  f3d849053f1390a09cada3fe172964a4  https://t.co/XL5AbJyr4m    False   

      Status  Subtype  Tags Thread Author      Thread Created Date  \
3625     NaN      NaN   NaN   ForosSemana  2020-10-22 21:01:31.000   

     Thread Entry Type            Thread Id  Thread URL  \
3625             share  1319383406493356032         NaN   

      Total Monthly Visitors  Twitter Author ID  Twitter Channel Role  \
3625              1414667000           18079284                   NaN   

      Twitter Followers  Twitter Following  Twitter Reply Count  \
3625            4563978                 53                    0   

     Twitter Reply to                                 Twitter Retweet of  \
3625              NaN  http://twitter.com/ForosSemana/statuses/131938...   

      Twitter Retweets  Twitter Tweets  Twitter Verified  Reach (new)  \
3625                 0          267874              True        77627   

      Reddit Score                                            mention  \
3625             0  ForosSemana,PorvenirOficial,RevistaDinero,Banc...   

           Dates      Time  
3625  2020-10-22  21:11:47

El usuario en cuestión solo realizó una mención y tiene un sentimiento neutral.

In [153]:
print(dfcopy["Hashtags"].value_counts())

#avancemosjuntos                                                                                                                            188
#generaciónporvenir, #elpoderdelasdecisiones, #avancemosjuntos                                                                              105
#academiadelahorro, #educaciónfinanciera, #avancemosjuntos                                                                                  101
#virtual7kporvenir, #avancemosjuntos                                                                                                         57
#transur                                                                                                                                     55
#porvenirvoyporti                                                                                                                            47
#pensiones                                                                                                                              